examples/rugby_ball/mapping_functions/Casper_mapping.f90

```diff
@@ -0,0 +1,178 @@
+  implicit none
+  private
+
+  !> A Casper mapping of coefficients 
+  !! Let's ask Casper for the real name of this mapping function...
+  !! but it was the one used in
+  !! https://doi.org/10.1002/fld.1964
+  !!
+  !! $f(x) = f_{min} + (f_{max} - f_{min}) x \frac{q + 1}{q + x}$ 
+  !!
+  !! It seems very similar to RAMP but with the convexity the other way
+  !!
+  !!  |       ...
+  !!  |    .. 
+  !!  |  .
+  !!  | . 
+  !!  |.
+  !!  |_________
+
+  type, public, extends(mapping_t) :: Casper_mapping_t
+  !> minimum value
+  real(kind=rp) :: f_min
+  !> maximum value
+  real(kind=rp) :: f_max
+  !> penalty parameter
+  real(kind=rp) :: q
+
+
+
+   contains
+     !> Constructor from json.
+     procedure, pass(this) :: init => Casper_mapping_init_from_json
+     !> Actual constructor.
+     procedure, pass(this) :: init_from_attributes => &
+          Casper_mapping_init_from_attributes
+     !> Destructor.
+     procedure, pass(this) :: free => Casper_mapping_free
+     !> Apply the forward mapping
+     procedure, pass(this) :: apply_forward => Casper_mapping_apply
+     !> Apply the adjoint mapping
+     procedure, pass(this) :: apply_backward => Casper_mapping_apply_backward
+  end type Casper_mapping_t
+
+contains
+
+  !> Constructor from json.
+  subroutine Casper_mapping_init_from_json(this, json, coef)
+    class(Casper_mapping_t), intent(inout) :: this
+    type(json_file), intent(inout) :: json
+    type(coef_t), intent(inout) :: coef
+
+    ! do the JSON stuff later
+    this%f_min = 0.0_rp
+    this%f_max = 1000.0_rp
+    this%q = 1.0_rp
+
+    call this%init_base(json, coef)
+    call Casper_mapping_init_from_attributes(this, coef)
+   
+  end subroutine Casper_mapping_init_from_json
+
+  !> Actual constructor.
+  subroutine Casper_mapping_init_from_attributes(this, coef)
+    class(Casper_mapping_t), intent(inout) :: this
+    type(coef_t), intent(inout) :: coef
+
+    ! there's actually nothing to do here.
+
+  end subroutine Casper_mapping_init_from_attributes
+
+  !> Destructor.
+  subroutine Casper_mapping_free(this)
+    class(Casper_mapping_t), intent(inout) :: this
+
+    call this%free_base()
+
+  end subroutine Casper_mapping_free
+
+  !> Apply the mapping
+  !! @param X_out mapped field
+  !! @param X_in unmapped field
+  subroutine Casper_mapping_apply(this, X_out, X_in)
+    class(Casper_mapping_t), intent(inout) :: this
+    type(field_t), intent(in) ::  X_in
+    type(field_t), intent(inout) ::  X_out
+
+    ! x_out = f_min + (f_max - f_min) * x_in * (q + 1) / (x_in + q) 
+    
+    ! TODO
+    ! Here we should make a descision, either use field_math for everything
+    ! or write individual backends.
+    !
+    ! I'm pro-field_math simply because these functions don't get executed too
+    ! often...
+    !
+    ! However! If we do descide to write backends for them, I (Harry) call
+    ! shotgun writing them, because they look easy and would be a good 
+    ! introduction to writing a kernel from scratch.
```

Comment:
Field math should be used if you want to do a single opperation, not when
chaining operations together.

Reply:
sticking with CPU for now